cmd/forward/main.go

```diff
@@ -226,9 +252,10 @@ func forwardQuery(ctx context.Context, queryState *presto.QueryStateInfo, client
 	}
 	successful, failed := atomic.Uint32{}, atomic.Uint32{}
 	forwardedQueries := sync.WaitGroup{}
+	cachedQueries := []QueryCache{}
```

Comment:
```suggestion
	cachedQueries := make([]QueryCache, len(clients)-1)
```
Pre-allocate the elements to prevent racing conditions.
See my comment below.